### PR TITLE
Feature/use common exception

### PR DESCRIPTION
**Pull request by hakuturu583**
## Types of PR

- [x] New Features
- [ ] Upgrade of existing features
- [ ] Bugfix

## Description

use common exception and remove old versions.

## How to review this PR.

please check pass all CI test.

### CODE REVIEW

File: simulation/traffic_simulator/src/traffic_lights/traffic_light_state.cpp [hunk 1]
```diff
@@ -38,10 +39,9 @@ std::istream & operator>>(std::istream & is, TrafficLightColor & datum)
   try {
     datum = conversions.at(value);
   } catch (const std::out_of_range &) {
-    std::stringstream what;
-    what << "An invalid value '" << value
-         << "' was specified for type 'traffic_simulator::TrafficLightColor'.";
-    throw std::runtime_error(what.str());
+    ;
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/src/traffic_lights/traffic_light_state.cpp:42]: Remove this line.

File: common/scenario_simulator_exception/include/scenario_simulator_exception/exception.hpp [hunk 2]
```diff
@@ -51,10 +53,28 @@ DEFINE_ERROR_CATEGORY(SemanticError);
...
+#define THROW_AUTOWARE_ERROR(...) \
+  THROW_ERROR(common::scenario_simulator_exception::AutowareError, __VA_ARGS__);
+#define THROW_SEMANTIC_ERROR(...) \
+  THROW_ERROR(common::scenario_simulator_exception::SemanticError, __VA_ARGS__);
+#define THROW_SIMULATION_ERROR(...) \
+  THROW_ERROR(common::scenario_simulator_exception::SimulationError, __VA_ARGS__);
+#define THROW_SPECIFICATION_VIOLATION(...) \
+  THROW_ERROR(common::scenario_simulator_exception::SpecificationViolation, __VA_ARGS__);
+#define THROW_SYNTAX_ERROR(...) \
+  THROW_ERROR(common::scenario_simulator_exception::SyntaxError, __VA_ARGS__);
```
Comment (yamacir-kit) [inline common/scenario_simulator_exception/include/scenario_simulator_exception/exception.hpp:72]: Semicolons should be added by the macro user and should not be included in the macro itself.

Comment (hakuturu583) [reply common/scenario_simulator_exception/include/scenario_simulator_exception/exception.hpp:72]: Thank you for pointing out! I will fix soon.

Comment (hakuturu583) [reply common/scenario_simulator_exception/include/scenario_simulator_exception/exception.hpp:72]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/306/commits/4e2999f700d892c18aaca045592630dc006ef69b

File: simulation/traffic_simulator/src/traffic_lights/traffic_light_state.cpp (hunk 1 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/src/traffic_lights/traffic_light_state.cpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/306/commits/8e359cbe365195565e1bbbfbc19663f990c859c1

**Review by yamacir-kit — APPROVED** [review]

**Pull request merged**